pkg/console/controllers/clidownloads/controller.go

```diff
@@ -0,0 +1,261 @@
+	queue        workqueue.RateLimitingInterface
+	recorder     events.Recorder
+}
+
+// factory func needs clients and informers
+func NewCLIDownloadsSyncController(
+	operatorConfigClient operatorclientv1.OperatorV1Interface,
+	operatorClient v1helpers.OperatorClient,
+	cliDownloadsInterface consoleclientv1.ConsoleCLIDownloadInterface,
+	routev1Client routeclientv1.RoutesGetter,
+	consoleCLIDownloadsInformers consoleinformersv1.ConsoleCLIDownloadInformer,
+	recorder events.Recorder,
+) *CLIDownloadsSyncController {
+
+	ctrl := &CLIDownloadsSyncController{
+		// configs
+		operatorConfigClient: operatorConfigClient.Consoles(),
+		// clients
+		operatorClient:            operatorClient,
+		consoleCliDownloadsClient: cliDownloadsInterface,
+		routeClient:               routev1Client,
+		recorder:                  recorder,
+		queue:                     workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ConsoleCliDownloadsSyncer"),
+	}
+
+	operatorClient.Informer().AddEventHandler(ctrl.newEventHandler())
+	consoleCLIDownloadsInformers.Informer().AddEventHandler(ctrl.newEventHandler())
+
+	ctrl.cachesToSync = append(ctrl.cachesToSync,
+		operatorClient.Informer().HasSynced,
+		consoleCLIDownloadsInformers.Informer().HasSynced,
+	)
+
+	return ctrl
+}
+
+func (c *CLIDownloadsSyncController) sync() error {
+	spec, _, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+	// only run if we are managed
+	if !management.IsOperatorManaged(spec.ManagementState) {
+		return nil
+	}
+
+	operatorConfig, err := c.operatorConfigClient.Get(api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	updatedOperatorConfig := operatorConfig.DeepCopy()
+
+	consoleRoute, err := c.routeClient.Routes(api.TargetNamespace).Get(api.OpenShiftConsoleRouteName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	host := routesub.GetCanonicalHost(consoleRoute)
+	ocConsoleCLIDownloads := PlatformBasedOCConsoleCLIDownloads(host, "amd64", ocCLIDownloadsCustomResourceName)
+	_, ocCLIDownloadsErrReason, ocCLIDownloadsErr := CreateOrUpdate(c.consoleCliDownloadsClient, ocConsoleCLIDownloads)
+	status.HandleDegraded(updatedOperatorConfig, "ConsoleCliDownloadsSync", ocCLIDownloadsErrReason, ocCLIDownloadsErr)
+	if ocCLIDownloadsErr != nil {
+		return ocCLIDownloadsErr
+	}
+
+	_, odoCLIDownloadsErrReason, odoCLIDownloadsErr := CreateOrUpdate(c.consoleCliDownloadsClient, ODOConsoleCLIDownloads())
+	status.HandleDegraded(updatedOperatorConfig, "ConsoleCliDownloadsSync", odoCLIDownloadsErrReason, odoCLIDownloadsErr)
+	if odoCLIDownloadsErr != nil {
+		return odoCLIDownloadsErr
+	}
+
+	return nil
+}
+
+func GetPlatformURL(baseURL, platform, archiveType string) string {
+	return fmt.Sprintf("%s/%s/%s", baseURL, platform, archiveType)
+}
+
+func PlatformBasedOCConsoleCLIDownloads(host, arch, cliDownloadsName string) *v1.ConsoleCLIDownload {
+	baseURL := fmt.Sprintf("%s/%s", util.HTTPS(host), arch)
+
```

Comment:
We could potentially have a map of strings here:

```golang
platforms := [string]string{
  "Linux": "linux",
   "Mac": "mac",
  "Windows": "windows",
}
```

and a loop to make the `Links`:

```golang
links := []v1.Link{}
for label, platform range platforms {
   links := append(links, Link{
      Href: GetPlatformURL(baseURL, platform, "oc.tar"),
      Text: fmt.Sprintf("Download oc for %v", Label)
   })
}
```
That or maybe just inline the `*PlatformKey`s since they are tied to the specifc "Download for *Platform*` text.

Reply:
Nice!